lib/stat_tracker.rb

```diff
@@ -13,42 +13,42 @@ def initialize(teams, games, game_teams)
         @game_teams = game_teams
     end
     
-    # def self.from_csv(data)
-    #     games = generate_csv_data(data[:games])
-    #     teams = generate_csv_data(data[:teams])
-    #     game_teams = generate_csv_data(data[:game_teams])
-
-    #     StatTracker.new(teams, games, game_teams)
-    # end
-
-    # def self.generate_csv_data(data)
-    #     csv_objects = []
-    #     CSV.foreach(data, headers: true) do |row|
-    #         if data == './data/test_games.csv'
-    #             csv_objects << Games.new(row)
-    #         elsif data == './data/teams.csv'
-    #             csv_objects << Teams.new(row)
-    #         elsif data == './data/test_game_teams.csv'
-    #             csv_objects << GameTeams.new(row)
-    #         end
-    #     end
-    #     csv_objects
-    # end
     def self.from_csv(data)
-        games = generate_csv_data(data[:games], Games)
-        teams = generate_csv_data(data[:teams], Teams)
-        game_teams = generate_csv_data(data[:game_teams], GameTeams)
+        games = generate_csv_data(data[:games])
+        teams = generate_csv_data(data[:teams])
+        game_teams = generate_csv_data(data[:game_teams])
 
         StatTracker.new(teams, games, game_teams)
     end
-    
-    def self.generate_csv_data(data, klass)
+
+    def self.generate_csv_data(data)
         csv_objects = []
         CSV.foreach(data, headers: true) do |row|
-            csv_objects << klass.new(row)
+            if data == './data/test_games.csv' || data == './data/games.csv'
+                csv_objects << Games.new(row)
+            elsif data == './data/teams.csv'
+                csv_objects << Teams.new(row)
+            elsif data == './data/test_game_teams.csv' || data == './data/game_teams.csv'
+                csv_objects << GameTeams.new(row)
```

Comment:
is this essentially hard-coding file path here to take in a file named either our fixture files or the spec harness?